### ISSUE TITLE
Unable to publish root content nodes that vary by culture.

### ISSUE THREAD

**Issue by meyntony** [umbraco/Umbraco-CMS#11123]
### Umbraco version

v9:rc003

### Bug summary

Publishing a root node that varies by culture, throws an exception in a fresh clean Umbraco installation.
![image](https://user-images.githubusercontent.com/85172999/133657595-cab48158-a112-49a2-856e-fb1022dc05a7.png)
```
Received an error from the server
An error occurred
Failed to get node at /.

Exception Details
System.Exception, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e: Failed to get node at /.
```

### Specifics

Here is the Stack trace:
```
at Umbraco.Cms.Infrastructure.PublishedCache.ContentCache.ApplyHideTopLevelNodeFromPath(IPublishedContent content, IList`1 segments, Boolean preview)
   at Umbraco.Cms.Infrastructure.PublishedCache.ContentCache.GetRouteByIdInternal(Boolean preview, Int32 contentId, Nullable`1 hideTopLevelNode, String culture)
   at Umbraco.Cms.Infrastructure.PublishedCache.ContentCache.<>c__DisplayClass13_0.<GetRouteById>b__0()
   at Umbraco.Extensions.AppCacheExtensions.<>c__DisplayClass5_0`1.<GetCacheItem>b__0()
   at Umbraco.Cms.Core.Cache.SafeLazy.<>c__DisplayClass1_0.<GetSafeLazy>b__0()
--- End of stack trace from previous location ---
   at Umbraco.Cms.Core.Cache.FastDictionaryAppCache.Get(String cacheKey, Func`1 getCacheItem)
   at Umbraco.Extensions.AppCacheExtensions.GetCacheItem[T](IAppCache provider, String cacheKey, Func`1 getCacheItem)
   at Umbraco.Cms.Core.Routing.RedirectTrackingHandler.StoreOldRoute(IContent entity, OldRoutesDictionary oldRoutes)
   at Umbraco.Cms.Core.Routing.RedirectTrackingHandler.StoreOldRoutes(IEnumerable`1 entities, IStatefulNotification notification)
   at Umbraco.Cms.Core.Routing.RedirectTrackingHandler.Handle(ContentPublishingNotification notification)
   at Umbraco.Cms.Core.Events.EventAggregator.PublishCore(IEnumerable`1 allHandlers, INotification notification)
   at Umbraco.Cms.Core.Events.EventAggregator.Publish[TNotification](TNotification notification)
   at Umbraco.Cms.Core.Events.ScopedNotificationPublisher.PublishCancelable(ICancelableNotification notification)
   at Umbraco.Cms.Core.Services.Implement.ContentService.StrategyCanPublish(IScope scope, IContent content, Boolean checkPath, IReadOnlyList`1 culturesPublishing, IReadOnlyCollection`1 culturesUnpublishing, EventMessages evtMsgs, IReadOnlyCollection`1 allLangs, IDictionary`2 notificationState)
   at Umbraco.Cms.Core.Services.Implement.ContentService.CommitDocumentChangesInternal(IScope scope, IContent content, EventMessages eventMessages, IReadOnlyCollection`1 allLangs, IDictionary`2 notificationState, Int32 userId, Boolean branchOne, Boolean branchRoot)
   at Umbraco.Cms.Core.Services.Implement.ContentService.SaveAndPublish(IContent content, String[] cultures, Int32 userId)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.PublishInternal(ContentItemSave contentItem, String defaultCulture, String cultureForInvariantErrors, Boolean& wasCancelled, String[]& successfulCultures)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.PostSaveInternal(ContentItemSave contentItem, Func`2 saveMethod, Func`2 mapToDisplay)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.PostSave(ContentItemSave contentItem)
   at lambda_method1530(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
```

I noticed that this issue didn't happen on my upgraded project from v9:rc002 to v9:rc003 (Not sure what to make of that), but a fresh v9:rc003 with a brand new document type that is **allowed as root** and **vary by culture** is a good candidate to reproduce this problem, its does not happen for child nodes either.

### Steps to reproduce

> This error was caused by allowing the root document type to be "Varied by Culture". I only have one language installed. I'm not sure if that is the problem, but this is easily replicated:
> Create a document type (empty), allow vary by culture, allow at root. Have only a single language installed. Create a root content of this type (don't have other content in the tree). Publish. You will see "Could not get the URL" in the links section of that content item and you will have the above error in the logs.

I've copied the steps to reproduce from @Shazwazza (as he noticed this first and reported in on slack), it can't get simpler than that.

### Expected result / actual result

Expected the root node to get published and get a Url

But an error was thrown, as shown above in the screenshot.

**Comment by bergmania** [comment]
I can reproduce in RC2 and RC3. Not RC1 - Apparently introduced in RC2

**Comment by JoseMarcenaro** [comment]
Maybe this helps narrow down the bug:

In RC3 (not in RC1) SurfaceControllers **do not have the default culture set** as they use to have in v8.
So, inside a `SurfaceController`, if you do something like this:
```
var docs = _umbracoHelper.ContentAtRoot() 
```
you only get the root documents that are _not culture variant_  - the others and invisible!
And even if you search a document by Id, i.e.  
```
var doc = _umbracoHelper.Content(1057)
```
you get a _null_ if the doc type is Culture Variant.

IMO this is a bug, not by design, right?  

It also happens in the other umbraco controllers - i.e. `UmbracoApiController`

**Comment by JoseMarcenaro** [comment]
(let me know if what I described above is a completely different problem - if so I will open a separate issue)

**Comment by bergmania** [comment]
Findings so far is `_variationContextAccessor.VariationContext` is null when referenced from backoffice .Seems to work from the website

**Comment by bergmania** [comment]
Thanks @JoseMarcenaro.. My PR also fixes the problem you describe. So your description makes me confident I found the bug.

**Comment by Shazwazza** [comment]
Fixed in #11124

**Comment by JoseMarcenaro** [comment]
Great! This is probably related to #10924 as well.
Thanks @bergmania.
